### PR TITLE
Updated: code to stop bopis functionality when the product is preorder or backorder(#1z55bnm)

### PR DESCRIPTION
**Pull request by ymaheshwari1**
- Added check for preorder/backorder
- Made the pickupstore property hidden by adding underscore(_)

### CODE REVIEW

File: scripts/script-tag.js [hunk 1]
```diff
@@ -95,6 +96,29 @@
     }
 
     // TODO: add preorder check
+    function isProductAvailableForBopis (virtualId, variantId) {
```
Comment (adityasharma7) [inline scripts/script-tag.js]:
```suggestion
    function isProductProrderedOrBackordered (virtualId, variantId) {
```

File: scripts/script-tag.js [hunk 2]
```diff
@@ -95,6 +96,29 @@
...
+                url: `${shopUrl}/admin/products/${virtualId}.json`,
+                crossDomain: true,
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                success: function (data) {
+                    if (data.product.tags.includes('Pre-Order') || data.product.tags.includes('Back-Order')) {
+                        resolve(data.product.variants.find((variant) => variant.id == variantId).inventory_policy === 'continue')
+                    }
+                    else
```
Comment (adityasharma7) [inline scripts/script-tag.js]: We should use {} for better readability

File: scripts/script-tag.js [hunk 3]
```diff
@@ -108,6 +132,8 @@
             // TODO Simplify this [name='id']. There is no need to serialize
             const cartForm = jQueryBopis("form[action='/cart/add']");
             const id = cartForm.serializeArray().find(ele => ele.name === "id").value;
+
+            if (await isProductAvailableForBopis(meta.product.id, id)) return;
```
Comment (adityasharma7) [inline scripts/script-tag.js]: Have you checked for error?

File: scripts/script-tag.js (hunk 1 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Sir, Changed the name.

File: scripts/script-tag.js (hunk 2 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Added block for else

File: scripts/script-tag.js (hunk 3 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Handled error using catch.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**